ksql-engine/src/main/java/io/confluent/ksql/util/KafkaConsumerGroupClientImpl.java

```diff
@@ -67,6 +88,20 @@ public KafkaConsumerGroupClientImpl(KsqlConfig ksqlConfig) {
     return results;
   }
 
+  private File flushPropertiesToTempFile() throws IOException {
+    File configFile = File.createTempFile("ksqlclient", "properties");
```

Comment:
Given this file contains credentials we need to be careful about the permissions on the created tmp file.

Ideally, we should be using encryption, but if we'll be switching to the Java API soon anyway, and assuming that does not require the tmp file, then it's probably sufficient for now to use  `java.nio.file.Files.createTempFile()` instead, which allows permissions to be restricted to stop other users reading the file.

Reply:
I updated the code to give only the owner read and write permissions on the file. As for encryption, the existing properties file already has credentials in unencrypted form. So this does not make anything less secure.